web/packages/teleport/src/JoinTokens/CreateJoinTokenDialog.tsx

```diff
@@ -0,0 +1,568 @@
+/**
+ * Teleport
+ * Copyright (C) 2024 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useState } from 'react';
+
+import {
+  Flex,
+  Text,
+  Box,
+  ButtonIcon,
+  ButtonText,
+  ButtonPrimary,
+  ButtonSecondary,
+  Alert,
+} from 'design';
+import styled from 'styled-components';
+import { HoverTooltip } from 'shared/components/ToolTip';
+import { Cross, Plus, Trash } from 'design/Icon';
+import Validation from 'shared/components/Validation';
+import FieldInput from 'shared/components/FieldInput';
+import { requiredField } from 'shared/components/Validation/rules';
+import {
+  FieldSelect,
+  FieldSelectCreatable,
+} from 'shared/components/FieldSelect';
+import { Option } from 'shared/components/Select';
+import { useAsync } from 'shared/hooks/useAsync';
+
+import { useTeleport } from 'teleport';
+import {
+  AWSRules,
+  CreateJoinTokenRequest,
+  JoinMethod,
+  JoinRole,
+  JoinToken,
+} from 'teleport/services/joinToken';
+
+const maxWidth = '550px';
+
+const joinRoleOptions: Option<JoinRole, JoinRole>[] = [
+  'App',
+  'Node',
+  'Db',
+  'Kube',
+  'Bot',
+  'WindowsDesktop',
+  'Discovery',
+].map(role => ({ value: role as JoinRole, label: role as JoinRole }));
+
+const availableJoinMethods: Option<JoinMethod, JoinMethod>[] = [
+  'iam',
+  'gcp',
+].map(method => ({
+  value: method as JoinMethod,
+  label: method as JoinMethod,
+}));
```

Comment:
Should the labels for these options be something a bit more user friendly?

Reply:
I chose this to match what's in the documentation and YAML for these join methods. I think in this case particularly this would be the most friendly/understood.